core/domain/auth_jobs_one_off.py

```diff
@@ -39,6 +39,258 @@
 ID_HASHING_FUNCTION = hash
 
 
+class SyncFirebaseAccountsOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to sync Firebase accounts with Oppia's user accounts.
+
+    This job does not print aggregated success output, because the reduce phase
+    is sharded by auth_id and could potentially generate tens of 1000s of
+    privacy-sensitive outputs. Instead, success is interpreted as an output
+    without any ERRORs.
+
+    This job is idempotent.
+
+    Pre-conditions:
+        - The models on Oppia reflect the most up-to-date values. Specifically,
+          a user is considered to "exist" IF AND ONLY IF a UserSettingsModel
+          exists.
+        - Exactly 1 FirebaseSeedModel exists.
+        - SeedFirebaseOneOffJob has succeeded AT LEAST ONCE in the history of
+          the database. This includes backups of the database created _after_
+          the job is run.
+        - PopulateFirebaseAccountsOneOffJob has succeeded AT LEAST ONCE in the
+          history of the database, WITHOUT creating any new accounts, AFTER
+          BOTH:
+              1. SeedFirebaseOneOffJob has succeeded.
+              2. Firebase user authentication has been deployed to the server.
+
+    Post-conditions:
+        - There is a 1-to-1-to-1 mapping between UserAuthDetailsModel,
+          UserIdByFirebaseAuthIdModel, and individual Firebase accounts; using
+          firebase_auth_id as the key. The sole exception is:
+              UserAuthDetailsModel(gae_id=feconf.SYSTEM_COMMITTER_ID)
+        - UserAuthDetailsModel.deleted,
+          UserIdByFirebaseAuthIdModel.deleted, and
+          firebase_admin.auth.ExportedUserRecord.disabled each have the same
+          respective boolean value.
+      """
+
+    SYSTEM_COMMITTER_ACK = 'INFO: SYSTEM_COMMITTER_ID skipped'
+
+    EMPTY_AUTH_ID_KEY = '<EMPTY_AUTH_ID_KEY>'
```

Comment:
This is a bit pedantic, but just confirming that "<" and ">" are disallowed chars in both Oppia and Firebase user IDs, right?

Reply:
Technically any ASCII character can be used as an auth ID, but our audit validators check that they're alphanumerics.